### PR TITLE
fix(treesitter): avoid `#` of `nil` value in _query_linter

### PR DESCRIPTION
**Pull request by Danielkonge**
I ran into the following error while editing/saving some query files:

```
Error executing lua callback: .../share/nvim/runtime/lua/vim/treesitter/_query_linter.lua:95: attempt to get length of a nil value
stack traceback:
	.../share/nvim/runtime/lua/vim/treesitter/_query_linter.lua:95: in function 'fn'
	/usr/local/share/nvim/runtime/lua/vim/func/_memoize.lua:54: in function 'parse'
	.../share/nvim/runtime/lua/vim/treesitter/_query_linter.lua:144: in function 'lint_match'
	.../share/nvim/runtime/lua/vim/treesitter/_query_linter.lua:182: in function 'fn'
	...l/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:489: in function 'for_each_tree'
	.../share/nvim/runtime/lua/vim/treesitter/_query_linter.lua:174: in function 'lint'
	/usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:804: in function 'lint'
	/usr/local/share/nvim/runtime/ftplugin/query.lua:28: in function </usr/local/share/nvim/runtime/ftplugin/query.lua:27>
```

This should fix that issue. 

Side note: This seems to be an older change, did anything in recent updates change something that might affect this? I updated neovim a few weeks ago and never ran into this, but after updating today I got the error. (Note: Before I build from brew, but that was broken for me today, so I built from source instead, which might make a difference?)

### CODE REVIEW

File: runtime/lua/vim/treesitter/_query_linter.lua [hunk 1]
```diff
@@ -92,7 +92,7 @@ local function get_error_entry(err, node)
     end_col = end_col + #underlined
   elseif msg:match('^Invalid') then
     -- Use the length of the problematic type/capture/field
-    end_col = end_col + #msg:match('"([^"]+)"')
+    end_col = end_col + #(msg:match('"([^"]+)"') or {})
```
Comment (lewis6991) [inline runtime/lua/vim/treesitter/_query_linter.lua:95]:
```suggestion
    end_col = end_col + #(msg:match('"([^"]+)"') or '')
```

Comment (Danielkonge) [reply runtime/lua/vim/treesitter/_query_linter.lua]: Yes, that makes more sense since `:match` returns `string | nil`. I will force-push an update.

**Comment by wookayin** [comment]
I've run into this problem before. Maybe the error was like:

> Invalid node type ""

see src/nvim/treesitter.c for where it error is thrown. Simply ignoring the nil result, resulting in offset 0 is a good enough fix, though.

Can you try printing out or inspecting what the `msg` string was exactly like?

**Comment by Danielkonge** [comment]
> I've run into this problem before. Maybe the error was like:
> 
> > Invalid node type ""
> > Invalid field name ""
> 
> see src/nvim/treesitter.c `query_err_to_string()` for where it error is thrown. Simply ignoring the nil result, resulting in offset 0 is a good enough fix, though.
> 
> Can you try printing out or inspecting what the `msg` string was exactly like? To see if there is any root cause (e.g. regression from the upstream tree-sitter library). A reproduction or the query file you were writing will be also helpful.

What would be the easiest way to inspect the message? The error was a typo, where I wrote something like
```query
(section_name
  "("
  ")")
```
instead of 
```query
(section_name
  "["
  "]")
```
in a `filename.scm` inside my `.config/nvim/after/queries/ini/` folder. I checked, and I get the same error in my `git_config` subfolder if I make a similar typo there. 

Usually I discover this kind of typo by using the diagnostics, e.g., in a Lua subfolder the query
```query
(bracket_index_expression
  "("
  ")")
```
gives diagnostics `Impossible pattern`, which I can then figure out is because the `(` and `)` should be `[` and `]`. But I can't get any diagnostics because of the error in the first example.

**Pull request merged**